src/obstacle_avoidance/src/MyNode.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <MyNode.h>
+
+sensor_msgs::LaserScan AvoidObstacleClass::lidar_data;
+
+int sizeOfLidarArray(sensor_msgs::LaserScan lidar);
+
+AvoidObstacleClass::AvoidObstacleClass() {
+  // initialise velocity
+  vel_msg.linear.x = LINEAR_VEL;
+  vel_msg.linear.y = 0;
+  vel_msg.linear.z = 0;
+  vel_msg.angular.x = 0;
+  vel_msg.angular.y = 0;
+  vel_msg.angular.z = 0;
+}
+
+AvoidObstacleClass::AvoidObstacleClass(int argc, char **argv,
+                                       std::string node_name) {
+  // Setup NodeHandles
+  ros::init(argc, argv, node_name);
+  nh = new ros::NodeHandle;
+
+  // Setup Subscriber/Publisher
+  twist_publisher = nh->advertise<geometry_msgs::Twist>("/cmd_vel", 10);
+  laser_scan_subscriber =
+      nh->subscribe("/scan", 10, &AvoidObstacleClass::lidarCallback, this);
+
+  // initialise velocity
+  vel_msg.linear.x = LINEAR_VEL;
+  vel_msg.linear.y = 0;
+  vel_msg.linear.z = 0;
+  vel_msg.angular.x = 0;
+  vel_msg.angular.y = 0;
+  vel_msg.angular.z = 0;
+  std::cout << "completed construction \n";
+  // std::cout << "lidar data " << lidar_data.angle_max << std::endl;
+  // std::cout << "lidar size " << sizeOfLidarArray(lidar_data) << std::endl;
+  // std::cout << lidar_data << std::endl;
+
+  /* Obtains character from the parameter server (or launch file), sets '!' as
+default
+std::string parameter_name = "my_node/character";
+std::string default_character = "!";
+SB_getParam(nh, parameter_name, suffix, default_character);
+
+// Setup Subscriber(s)
+std::string topic_to_subscribe_to = "subscribe_topic";
+int refresh_rate = 10;
+my_subscriber = nh.subscribe(topic_to_subscribe_to, refresh_rate,
+&MyClass::subscriberCallBack, this);
+
+// Setup Publisher(s)
+std::string topic = private_nh.resolveName("publish_topic");
+uint32_t queue_size = 1;
+my_publisher = private_nh.advertise<std_msgs::String>(topic, queue_size);
+  */
```

Comment:
In general, commented out code should be deleted before finalising your code. Try not to worry too much about losing it,  one of the joys of git is that nothing that is committed ever really vanishes.